src/renderer/api/catalog-entity-registry.ts

```diff
@@ -27,15 +27,25 @@ import type { Cluster } from "../../main/cluster";
 import { ClusterStore } from "../../common/cluster-store";
 import { Disposer, iter } from "../utils";
 import { once } from "lodash";
+import type { CatalogEntityItem } from "../../renderer/components/+catalog/catalog-entity-item";
 
 export type EntityFilter = (entity: CatalogEntity) => any;
+export type CatelogEntityOnRunHook = (entity: CatalogEntity | CatalogEntityItem<CatalogEntity>) => boolean | Promise<boolean>;
+
+type CatalogEntityUid = CatalogEntity["metadata"]["uid"];
 
 export class CatalogEntityRegistry {
   @observable protected activeEntityId: string | undefined = undefined;
   protected _entities = observable.map<string, CatalogEntity>([], { deep: true });
   protected filters = observable.set<EntityFilter>([], {
     deep: false,
   });
+  protected entityOnRunHooks = observable.set<{
+    catalogEntityUid: CatalogEntityUid;
+    onRunHook: CatelogEntityOnRunHook
+  }>([], {
+    deep: false,
+  });
```

Comment:
This is a very strange type, why is it not `observable.map<string, CatalogEnitytOnRunHook[]>([], { deep: false });`?

Reply:
We need `catalogEntityUid` because there are `getOnRunHook `

Reply:
The top level map's key would be `CatalogEntityUid`

Reply:
`string` is not as obvious as `catalogEntityUid`, but again, just my opinion, maybe you know more about the convention in Lens.

Reply:
We don't really have a convention... (namely we use both styles). What I meant was I should have suggested the type `observable.map<CatalogEntityUid, IObservableSet<CatelogEntityOnRunHook>>`

Reply:
OK, thanks for clarify.